main.cmake

```diff
@@ -44,6 +45,12 @@ option(MENDER_ARTIFACT_GZIP_COMPRESSION "Enable GZIP compression support when do
 option(MENDER_ARTIFACT_LZMA_COMPRESSION "Enable LZMA compression support when downloading and extracting Artifacts (Default: ON)" ON)
 option(MENDER_ARTIFACT_ZSTD_COMPRESSION "Enable Zstd compression support when downloading and extracting Artifacts (Default: ON)" ON)
 
+find_program(CCACHE ccache)
+if(CCACHE)
```

Comment:
I'm a little bit uneasy about adding it as the default. I've been using it for a long time already (by specifying `-D CMAKE_CXX_COMPILER_LAUNCHER=ccache`), and most of the time it's been fine. But I have had problems in few weird instances, particularly when using code coverage, where blowing out the entire cache was the only thing that worked. This is rather unexpected if it's the default and you don't know that ccache is being used.

Reply:
Agreed. I can wrap this in if(BUILD_TESTS), but it may be worth looking into if tests are failing because ccache is enabled.

Reply:
I too am not a fan of adding this as the default. I've run into some issues with `ccache` myself.

Reply:
Mostly tests, but I'm still skeptical of it. I see that `Gentoo` recommends against it: https://wiki.gentoo.org/wiki/Handbook:AMD64/Working/Features#Caching_compilation_objects. 

But it's really nice to have the option to enable it though I think.